coalib/bears/Bear.py

```diff
@@ -305,6 +310,45 @@ def log_message(self, log_message, timestamp=None, **kwargs):
     def run(self, *args, dependency_results=None, **kwargs):
         raise NotImplementedError
 
+    def setup_profile(self, prof=None):
+        path = os.getcwd()
+        if self.profile == 'True':
+            filename = '{}_{}.prof'.format(self.section.name, self.name)
+            open(filename, 'wb')
+            prof.dump_stats(filename)
+        else:
+            subdir = self.profile
+            filepath = os.path.join(path, subdir)
+            if os.path.isdir(filepath):
+                filename = '{}_{}.prof'.format(self.section.name, self.name)
+                open(os.path.join(filepath, filename), 'wb')
+                prof.dump_stats(os.path.join(filepath, filename))
+
+            else:
+                os.mkdir(os.path.join(path, subdir))
+                filename = '{}_{}.prof'.format(self.section.name, self.name)
+                open(os.path.join(filepath, filename), 'wb')
+                prof.dump_stats(os.path.join(filepath, filename))
+
+    def profile_run(self, *args, **kwargs):
+        prof1 = cProfile.Profile()
```

Comment:
please name them properly ;) proper naming automatically makes code documented :)

Reply:
Why not actually running everything in the same profile object?